### PR TITLE
[Tooling] Add Buildkite annotation on test failures — DRY using bash-cache plugin

### PR DESCRIPTION
**Pull request by AliSoftware**
This builds on top of https://github.com/wordpress-mobile/WordPress-iOS/pull/18952 to generalize the approach by moving the logic into the `bash-cache-buildkite-plugin`

> **Warning** This PR depends on https://github.com/Automattic/bash-cache-buildkite-plugin/pull/27 to be merged first. Once the PR in bash-cache repo lands, we need to release a new version of the plugin and update its reference in this PR's `&common_plugins` anchor.

### CODE REVIEW

File: .buildkite/pipeline.yml
Comment (AliSoftware) [inline .buildkite/pipeline.yml]: 🚨  This needs to be updated to the new version of `bash-cache` once we land https://github.com/Automattic/bash-cache-buildkite-plugin/pull/27 and make a new version of the plugin.

**Comment by AliSoftware** [comment]
[Works as expected](https://buildkite.com/automattic/wordpress-ios/builds/8640#_) 🎉 

image

**Comment by AliSoftware** [comment]
Oh, [look, a flaky test failure](https://buildkite.com/automattic/wordpress-ios/builds/8643#0181af51-7cc2-41d9-882c-53ca7a2961d1)! 🙂 
image

**Comment by AliSoftware** [comment]
Ok, my next step after adding latest commits to the plugin (up to [9c9b755](https://github.com/Automattic/bash-cache-buildkite-plugin/pull/27/commits/9c9b755d001d21bd71a7229eeb14222336b235de)) to remove annotations on retry, was to:

1. run the test once and make them flaky-fail, ensure the annotation got added, then
2. hit "Retry" on a failed step and ensure that the annotation got removed once the second try ultimately pass.

I've done step 1, and the corresponding Buildkite job indeed had some failures — one of them that I created intentionally by connecting on the VM via ssh while the UI Tests were running, then killing `wiremock` to make the test fail:
image

Though notice that:
 - There were 2 annotations ("Unit Tests" and "UI Tests (iPhone)")
 - But only the "Unit Tests" job is marked in red and as failed in the summary
 - This is likely because, for the "UI Tests (iPhone)" step, it seems that there was some internal retry mechanism that got kicked… yet the `report.junit` still kept the 2 failed tests marked as failures, while the logs suggests that there were failures but ends with "Tests succeeded"…

```
  | [20:10:54]: ▸ WordPressUITests.EditorGutenbergTests
  | [20:10:54]: ▸   testAddGalleryBlock, failed: caught error: "timedOut"
  | [20:10:54]: ▸   <unknown>:0
  | [20:10:54]: ▸   testAddGalleryBlock, Failed to get automation session for org.wordpress:5810: Timed out while requesting automation session for org.wordpress (pid:5810).
  | [20:10:54]: ▸   <unknown>:0
  | [20:10:54]: ▸ 	 Executed 16 tests, with 1 test skipped and 2 failures (1 unexpected) in 1997.767 (1997.789) seconds
  | [20:10:54]: ▸
  | [20:11:13]: ▸ 2022-06-29 20:11:13.373 xcodebuild[5355:14972] [MT] IDETestOperationsObserverDebug: 2098.462 elapsed -- Testing started completed.
  | [20:11:13]: ▸ 2022-06-29 20:11:13.373 xcodebuild[5355:14972] [MT] IDETestOperationsObserverDebug: 0.000 sec, +0.000 sec -- start
  | [20:11:13]: ▸ 2022-06-29 20:11:13.373 xcodebuild[5355:14972] [MT] IDETestOperationsObserverDebug: 2098.462 sec, +2098.462 sec -- end
  | [20:11:13]: ▸ Test execute Succeeded
  | +--------------------+--------------------+
  | |              Test Results               |
  | +--------------------+--------------------+
  | | Number of tests    | 14 (and 2 retries) |
  | | Number of failures | 0                  |
  | +--------------------+--------------------+
```

I don't think this is really related to the changes in this PR though (but rather an odd behavior/bug of `xcodebuild` / `fastlane` + `xcpretty` reporting 2 failures in the logs and in the junit report… but 0 failures in the summary and exit status). So I'm not considering fixing that (after all this PR does the job we expect of it, which is transform the failures from the junit report into corresponding annotations, and if the junit report is incorrect it's not really the job of this PR to fix)

---

I'll now hit "Retry" on the steps and see if it appropriately remove the previous annotations when the reruns ultimately go green.

**Comment by AliSoftware** [comment]
> I'll now hit "Retry" on the steps and see if it appropriately remove the previous annotations when the reruns ultimately go green.

[The re-run of the pseudo-flaky failed "Unit Tests" step](https://buildkite.com/automattic/wordpress-ios/builds/8650#_) properly removed the previous annotations once the rerun passed 🎉 

image

**Review by mokagio — APPROVED** [review]
Approving to unblock. The only thing left to do is merging the plugin PR, pushing a new tag with it, and using it here (as you [noted](https://github.com/wordpress-mobile/WordPress-iOS/pull/18961/files#r909459386))

Update: Added "[Status] DO NOT MERGE" as a precaution.

File: .buildkite/pipeline.yml
Comment (AliSoftware) [reply .buildkite/pipeline.yml]: Released [bash-cache 2.6.0](https://github.com/Automattic/bash-cache-buildkite-plugin/releases/tag/2.6.0), and updated the plugin version reference accordingly here in 189002dd6b

**Pull request merged**
